src/hooks/useFavouriteMessages.ts

```diff
@@ -15,27 +14,65 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+import { MatrixEvent } from "matrix-js-sdk/src/models/event";
 import { useState } from "react";
 
-const favouriteMessageIds = JSON.parse(
-    localStorage?.getItem("io_element_favouriteMessages")?? "[]") as string[];
+interface IButtonProp {
+    mxEvent?: MatrixEvent;
+}
+
+let sortState = false;
+let isSearchClicked = false;
+
+export default function useFavouriteMessages(props?: IButtonProp) {
+    const favouriteMessagesIds = JSON.parse(
+        localStorage?.getItem("io_element_favouriteMessages")?? "[]") as any[];
```

Comment:
```suggestion
        localStorage?.getItem("io_element_favouriteMessages") ?? "[]") as any[];
```

Why is `as any[]` required?

Reply:
Just tryna get pass typescript's typechecking and for easier access to array methods.